### ISSUE TITLE
Provider produced inconsistent final plan for azurerm_eventgrid_topic

### ISSUE THREAD

**Issue by zerovirus999** [hashicorp/terraform-provider-azurerm#19921]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.27

### Affected Resource(s)/Data Source(s)

azurerm_eventgrid_topic

### Terraform Configuration Files

```hcl
resource "azurerm_eventgrid_topic" "change_event_topic" {
  name = "<whateverYouLike>"

  resource_group_name = azurerm_resource_group.app_resource_group.name
  location            = azurerm_resource_group.app_resource_group.location

  input_schema = "CloudEventSchemaV1_0"

  local_auth_enabled = false

  public_network_access_enabled = local.env.is_feature_test

  dynamic "inbound_ip_rule" {
    for_each = local.env.is_feature_test ? module.docker_app_service[0].outbound_ip_address_list : []

    content {
      ip_mask = inbound_ip_rule.value
      action = "Allow"
    }
  }

  tags = module.bsk_global.tags
}


is_feature_test is set to True in the this case. 

I'm trying to include the bare minimum without having to violate any compliance policies. 
We tried using the latest release as well  3.28. This was working fine with version 3.27 for the until Friday when version 3.28 came out. Not sure how the latest release breaks the previous one, but that that's the only change I'm aware of. The Code itself on our end never changed.

EDIT: It will fail the first time giving the error listed below. Then a subsequent run fixes it. Our Pipeline clears the resources once it's provisions the resources and is done testing. So when we re-run it again it will fail and re-run works again. The cycle continues like this.
```

### Debug Output/Panic Output

```shell
-
```

### Expected Behaviour

Should have created an event grid topic

### Actual Behaviour

**Error: Provider produced inconsistent final plan

When expanding the plan for azurerm_eventgrid_topic.change_event_topic to

include new values learned so far during apply, provider

"registry.terraform.io/hashicorp/azurerm" produced an invalid new value for

.inbound_ip_rule: new element 1 has appeared.

This is a bug in the provider, which should be reported in the provider's own

issue tracker.**

### Steps to Reproduce

terraform apply -auto-approce -input=plan.tfplan

**Comment by xiaxyi** [comment]
Thanks @zerovirus999 for raising this issue, if my understanding is correct, are you suggesting that you are not able to run the TF config for the first time but able to run it in the next apply?

**Comment by zerovirus999** [comment]
> Thanks @zerovirus999 for raising this issue, if my understanding is correct, are you suggesting that you are not able to run the TF config for the first time but able to run it in the next apply?

Yeah, that is correct. The First apply fails with the listed error. The way we are doing a terraform apply is using TFPlans. So, once the resources are cleaned up, we get the same error again. Re-run and it works fine. This started happening on Friday moments after version 3.28 was release for the provider. At first I thought it was the latest provider since we are using a ~>3.24 constraint for the version, but the Terraform Init output shows that it's downloading v3.27.

**Comment by zerovirus999** [comment]
> Thanks @zerovirus999 for raising this issue, if my understanding is correct, are you suggesting that you are not able to run the TF config for the first time but able to run it in the next apply?

Hi, is there an update on this?

**Comment by xiaxyi** [comment]
Thanks @zerovirus999 for the update, I tried terraform 3.37 to deploy the event grid topic with IP_rules and it was deployed successfully, then I used 3.28 to plan/apply and I didn't get the error you were seeing. To better narrow down the error, can you try testing whole process using the direct assignment of ip rules, instead of using the dynamic block?

**Comment by rcskosir** [comment]
Thanks for opening this issue. Since time has passed and @xiaxyi was not able to reproduce this issue, I am going to mark this issue closed as “can’t reproduce”. If you are able to provide more information to help us reproduce this issue we may re-open this issue or open a new issue, thanks!
